### ISSUE TITLE
Preserve link IDs when possible

### ISSUE THREAD

**Issue by Beanow** [msrd0/cargo-doc2readme#367]
As a small readability feature when looking at the output markdown, it'd be nice if existing IDs in docs like:

```md
[Have you a link][some-link-id]

[some-link-id]: https://github.com/msrd0/cargo-doc2readme
```

Were preserved and didn't turn into:
```md
[Have you a link][__link0]

[__link0]: https://github.com/msrd0/cargo-doc2readme
```

It looks like this would depend on https://github.com/raphlinus/pulldown-cmark/issues/434
Which got resolved, but is not yet released in a new crate version (seems planned to be `v0.10.0`).
https://github.com/raphlinus/pulldown-cmark/blob/80eeca9ca87b55e5114db019b93b50aadc2a66cf/src/lib.rs#L187-L188

**Comment by msrd0** [comment]
It looks like this is blocked on a new release from pulldown-cmark.

**Comment by msrd0** [comment]
We are using pulldown-cmark >=0.10 for some time now. If someone wants to implement this, and the code changes required are minimal, I'm happy to review a PR. Otherwise, note that _verbatim copy of your markdown_ is a non-goal of this project, since a readable version of the same markdown is already present, and this tool is only here to put that markdown somewhere websites like crates.io, codeberg.org, github.com etc can find and present it.
